app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Oneoffs::NPQ
+  class BackfillTrnToTeacherProfile
+    def migrate
+      # locate all TeacherProfiles with empty TRN
+      incorrect_teacher_profiles.in_batches.each_record do |profile|
+        trns = profile.user.npq_applications.map(&:teacher_reference_number).uniq
+        # check if they are correct
+        if trns.count == 1 && trns.first.length == 7
```

Comment:
So we have this to help ensure that only valid, correctly formatted TRNs are stored on the `TeacherProfile` - could that be used here perhaps given you're checking?